### PR TITLE
add anchor info for the warning message of invalid link

### PR DESCRIPTION
**Pull request by ansyral**
@vwxyzh @chenkennt @hellosnow @superyyrrzz @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 1]
```diff
@@ -224,6 +224,7 @@ public MarkupResult ParseHtml(string html, FileAndType ft)
                     sources.Add(new LinkSourceInfo
                     {
                         Target = file,
+                        Anchor = anchor,
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >anchor [](start = 33, length = 6)

Note whether we need to remove anchor info in RemoveDebugInfo, which just added by Zhenghui?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs (hunk 1 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: this commit only add the anchor info into LinkSourceInfo, not add an attribute into the doc

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: Finally find it's introduced in [this](https://github.com/dotnet/docfx/commit/051548b6ecf6b4469a2dc56951154545ba309593) commit... This anchor is an intermediate attribute used in resolving xref, and has been removed [here](https://github.com/dotnet/docfx/commit/051548b6ecf6b4469a2dc56951154545ba309593#diff-7bc60e42b31ac047d2df1c6b05f100d5R450).

**Pull request merged**